pkg/filters/fields.go

```diff
@@ -81,15 +81,51 @@ type FieldFilter struct {
 	fields         fmutils.NestedMask
 	action         tetragon.FieldFilterAction
 	invertEventSet bool
+	needsCopy      map[tetragon.EventType]struct{}
 }
 
 // NewFieldFilter constructs a new FieldFilter from a set of fields.
 func NewFieldFilter(eventSet []tetragon.EventType, fields []string, action tetragon.FieldFilterAction, invertEventSet bool) *FieldFilter {
+	var maybeNeedsCopy bool
+	if action == tetragon.FieldFilterAction_EXCLUDE {
+		for _, field := range fields {
+			if field == "process.pid" || field == "parent.pid" {
```

Comment:
Can you explain this check? Why is it only "process.pid" that we care about? Shouldn't be the same with all the fields under process or parent? I would expect something like strings.HasPrefix(field, "process.") || strings.HasPrefix(field,  "parent.").

Reply:
The bug is specifically related to missing pids causing segfaults in the eventcache. If these fields aren't filtered, we don't hit the bug. Maybe it is safer to do just process and parent in general though.

Reply:
OK, but there might other similar bugs. Is there an argument we can make that the same will not happen for other fields in process?

My understanding of the issue is that the fieldfilter will modify .process and .parent of the event which are shared among different processes. So a fieldfilter applied to one event, might modify .process in a way that affects another event. Based on above reasoning, if a fieldFilter, for example, deletes process.pod for one event, it will also affect other types of events. Maybe I'm not understanding the issue correctly, but I'd like us to have an argument that something similar cannot happen for other fields under .process (and similarly .parent).

Let me also CC @tpapagian in case he has any thoughts.

Reply:
Actually you're right. Managed to get a different bug with another fieldfilter config. I've updated accordingly.